src/world/daily_tally.cpp

```diff
@@ -30,7 +30,7 @@ namespace dailytally
 {
     void UpdateDailyTallyPoints()
     {
-        uint16 dailyTallyLimit  = settings::get<uint16>("main.DAILY_TALLY_LIMIT");
+        uint32 dailyTallyLimit  = settings::get<uint16>("main.DAILY_TALLY_LIMIT");
```

Comment:
Shouldn't this be int32?

Reply:
The limit doesn't or shouldn't actually support negative numbers, it just needs to be big enough to hold a uint16 in a 32 bit space. Either should work in this case so whatever is preferred.

Reply:
-1 on any uint should be 2^bitsize-1 

It would have the same problem logically, i'm guessing the compiler (or the connector?) does some funky stuff with these prepared statements.

I would promote both`dailyTallyLimit` and `dailyTallyAmount` to int32, use `settings::get<int32>` but clamp to 0: `std::max<int32>(0, settings::get<int32>("main.DAILY_TALLY_AMOUNT")));`

Reply:
We don't want to actually exceed 16 bits though, since that's what we send in the packet. I could put it in a int32 and clamp it to uint16 numeric limits, but that's basically already what it's doing (settings::get casts it to a uint16). The issue is that any sufficiently large uint16 is interpreted as a negative int16. If we use a 32 bit setting then we would have the same problem when the 32 sign bit starts getting flipped (though packet issues would arise before that).

Reply:
```
        const auto dailyTally = rset->get<int32>("daily_tally");
        ref<uint16>(0x5E)     = dailyTally == -1 ? 0 : static_cast<uint16>(dailyTally);
```
Looks like it's doing what I would expect because we use -1 tally as a default
db is stored as signed, and type is used as a downcasted uint16

Reply:
I hope this is okay. I clamped it to uint16 range since that's the valid packet range, but use an int32 to query the db so all types should match up.

Reply:
Ooh, I love that. That's super cool and very obvious on what the intent is

Reply:
I feel like there was a reason I used -1 instead of null, but I can't remember now. Maybe could make this all uint16 and just use null instead of -1, will have to look into that eventually.

Reply:
That would be ideal, but we'd need a migration most likely. Maybe it's doable?

Reply:
It looks like `CLuaBaseEntity::getCurrency` and in turn `charutils::GetPoints` are what's ultimately preventing null from being used here. Only other thing I can think of would be using something else to indicate whether they've done the into or not like a charvar but then you have a bunch of forever vars which seems worse than the current implementation. This works now anyway so just gonna leave as is.